setting `prerender: false` creates two closing `</head>` tags in a template with a `<style>` tag

## Type of Change
- [ ] New Feature Request
- [ ] Documentation / Website
- [ ] Improvement / Suggestion
- [x] Bug
- [ ] Other (please clarify below)

As part of working on #855 , I noticed that the _Default Greenwood Configuration and Workspace w/Custom Page Template_ specs were failing due to an [error in the smoke test of mismatching `<head>` tags](https://github.com/ProjectEvergreen/greenwood/pull/857/files#diff-28cb82df7ef31ace541ab9f325985abd46451f4b22b6176e39f69fbada68729eR53).
```
  2) Running Common Index Smoke Tests for Default Greenwood Configuration and Workspace w/Custom Page Template
       document <head>
         should have matching opening and closing <head> tags in the <head>:

      AssertionError: expected false to equal true
      + expected - actual

      -false
      +true
```

## Details
Seems similar to #810 and something about the proximity / usage of a `<style>` tag?  Here is that the final generated HTML looks like
```html
<!DOCTYPE html>
  <html lang="en" prefix="og:http://ogp.me/ns#">



  <head>
    <link rel="preload" href="/styles/theme.16214223.css" as="style" crossorigin="anonymous">
    </link>

    <link rel="modulepreload" href="/main.afc4433b.js" as="script">




    <title>My App</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black" />

    <meta property="og:description" content="My custom meta content." />



    <script type="module" src="/main.afc4433b.js"></script>

    <link rel="stylesheet" href="/styles/theme.16214223.css" />
  </head>
  <style>
    .owen-test {
      color: blue;
    }

    h3 {
      color: green;
    }
  </style>
  </head>




  <body>


    <div class='wrapper'>
      <div class='page-template content owen-test'>
        <h3>Hello</h3>

        <h1>Welcome to Greenwood!</h1>

      </div>
    </div>


  </body>

</html>
```

--- a/packages/cli/src/plugins/resource/plugin-standard-html.js
+++ b/packages/cli/src/plugins/resource/plugin-standard-html.js
@@ -177,7 +177,11 @@ const getAppTemplate = (contents, templatesDir, customImports = [], contextPlugi
 
     // merge <style> tags
     if (headStyles.length > 0) {
-      const latestHeadContents = appTemplateContents.match(/<head>.*.<\/head>/s)[0];
+      const matchNeedleStyle = /<style .*/g;
+      const appHeadStyleMatches = appTemplateHeadContents.match(matchNeedleStyle);
+      const lastStyle = appHeadStyleMatches && appHeadStyleMatches.length && appHeadStyleMatches.length > 0
+        ? appHeadStyleMatches[appHeadStyleMatches.length - 1]
+        : '</head>';
       const pageBodyStyles = headStyles.map((style) => {
         const attributes = style.rawAttrs === '' ? '' : ` ${style.rawAttrs}`;
 
@@ -187,13 +191,17 @@ const getAppTemplate = (contents, templatesDir, customImports = [], contextPlugi
           </style>
         `;
       });
-      const lastIndex = latestHeadContents.lastIndexOf('</style>') === -1
-        ? latestHeadContents.lastIndexOf('</head>')
-        : latestHeadContents.lastIndexOf('</style>');
-      const offset = lastIndex === -1 ? '</head>'.length : '</style>'.length;
-      const result = latestHeadContents.substring(0, lastIndex + offset) + pageBodyStyles.join('\n') + latestHeadContents.substring(lastIndex);
 
-      appTemplateContents = appTemplateContents.replace(latestHeadContents, result);
+      if (lastStyle === '</head>') {
+        appTemplateContents = appTemplateContents.replace(lastStyle, `
+          ${pageBodyStyles.join('\n')}
+        ${lastStyle}\n
+      `);
+      } else {
+        appTemplateContents = appTemplateContents.replace(lastStyle, `${lastStyle}\n
+          ${pageBodyStyles.join('\n')}
+        `);
+      }
     }
 
     // merge <meta>

--- a/packages/cli/test/cases/build.default.workspace-template-page/build.default.workspace-template-page.spec.js
+++ b/packages/cli/test/cases/build.default.workspace-template-page/build.default.workspace-template-page.spec.js
@@ -86,8 +86,8 @@ describe('Build Greenwood With: ', function() {
           expect(linkTags.length).to.equal(1);
         });
 
-        it('should have 2 <style> tags in the <head> (1 + one from Puppeteer)', function() {
-          expect(styleTags.length).to.equal(2);
+        it('should have 1 <style> tags in the <head>', function() {
+          expect(styleTags.length).to.equal(1);
         });
 
         it('should add one page template <script> tag', function() {
@@ -101,8 +101,7 @@ describe('Build Greenwood With: ', function() {
         });
 
         it('should add one page template <style> tag', function() {
-          // offset index by one since first <style> tag is from Puppeteer
-          expect(styleTags[1].textContent).to.contain('.owen-test');
+          expect(styleTags[0].textContent).to.contain('.owen-test');
         });
       });
   

--- a/packages/cli/test/cases/build.default.workspace-template-page/greenwood.config.js
+++ b/packages/cli/test/cases/build.default.workspace-template-page/greenwood.config.js
@@ -1,0 +1,3 @@
+export default {
+  prerender: false
+};

